### PR TITLE
fix: validate against blank passwords

### PR DESCRIPTION
**Pull request by flowck**
> I am open to contributions. If you want to land a hand, please open an issue or a pull request. I will work on the contribution guidelines soon.

Your note was very welcoming, so... I thought I could start somewhere. 

### Changes

- Updated the request validator functions to account for passwords with blank characters such as `        `;
- Added test cases to assert against blank passwords;

### Potential follow up

- The test cases in the `request` package are good candidates to be written in a table-driven manner given how simple they are to setup.

### CODE REVIEW

File: apps/core/internal/request/create_user_request_test.go [hunk 1]
```diff
@@ -26,6 +26,14 @@ func TestValidateCreateUserRequest(t *testing.T) {
...
 
+	t.Run("it should return an error if the password is made of empty spaces", func(t *testing.T) {
+		t.Parallel()
+
+		req := request.CreateUserRequest{Email: "mail@domain.tld", Password: "        "}
+		err := req.Validate()
+		require.Error(t, err)
+	})
+
 	t.Run("it should validate the request", func(t *testing.T) {
```
Comment (arthurdotwork) [inline apps/core/internal/request/create_user_request_test.go:37]: Same as above!

File: apps/core/internal/request/authenticate_request_test.go [hunk 2]
```diff
@@ -26,6 +26,14 @@ func TestValidateAuthenticationRequest(t *testing.T) {
...
 	})
 
+	t.Run("it should return an error if the password is made of empty spaces", func(t *testing.T) {
+		t.Parallel()
+
+		req := request.AuthenticateRequest{Email: "mail@domain.tld", Password: "        "}
+		err := req.Validate()
+		require.Error(t, err)
+	})
+
```
Comment (arthurdotwork) [inline apps/core/internal/request/authenticate_request_test.go:36]: I believe we could add this in the test above with "short____" password, would cover the same amount of code. 👍

**Review by arthurdotwork — COMMENTED** [review]
Thanks for the contribution, just two little comments on tests.

For the test table, I'm not really fond of it so I'd rather not go in that direction 😅

File: apps/core/internal/request/create_user_request_test.go (hunk 1 above)
Comment (flowck) [reply apps/core/internal/request/create_user_request_test.go]: @arthurdotwork This should do [639a5cd](https://github.com/arthurdotwork/dreamtrader/pull/16/commits/639a5cdf3d6fb4bf6457d05a017923d547ea9d61)

File: apps/core/internal/request/authenticate_request_test.go (hunk 2 above)
Comment (flowck) [reply apps/core/internal/request/authenticate_request_test.go]: @arthurdotwork This should do [639a5cd](https://github.com/arthurdotwork/dreamtrader/pull/16/commits/639a5cdf3d6fb4bf6457d05a017923d547ea9d61)

**Pull request closed without merging**